tests/admin_scripts/tests.py

```diff
@@ -2234,10 +2259,32 @@ def test_basic(self):
         self.assertOutput(out, "FOO = 'bar'  ###")
 
     def test_settings_configured(self):
+        self.write_configured_settings(args='DEBUG=True')
+        self.addCleanup(self.remove_configured_settings)
         out, err = self.run_manage(['diffsettings'], configured_settings=True)
         self.assertNoOutput(err)
         self.assertOutput(out, 'DEBUG = True')
 
+    def test_dynamic_settings_configured(self):
+        self.write_configured_settings(
+            extra='''class Settings:
```

Comment:
The purpose of this class isn't obvious to me. Is this code adapted your use case? Maybe you can add a bit more explanation to the ticket. I'm not sure what "custom default settings" are.

Reply:
It's not obvious to me that the template approach is the best solution for readability as opposed to just creating another test settings file.

Reply:
It is somewhat. All that's needed is an object with an attribute that doesn't appear or has its value different from the default and is listed in the dir().

By default settings I meant the default_settings argument passed to settings.configure(). I tried to follow the terminology from the [settings docs](https://docs.djangoproject.com/en/dev/topics/settings/#custom-default-settings).